docs/cep/examples/source-and-sink-example.md

```diff
@@ -0,0 +1,53 @@
+---
+sidebar_position: 4
+title: Source and Sink Example
+---
+
+Sources and sinks are used to consume and publish events to external systems.
+
+There are multiple source and sink types, but this example only explains Macrometa source with a c8stream sink. For more info refer to the [Stream Worker Query Guide](../query-guide/index.md).
+
+## Example
+
+This example creates a source from which a stream consumes JSON messages:
+
+```sql
+C8DB source to consume `JSON` messages from.
+CREATE SOURCE TemperatureStream WITH (type='database', collection='TemperatureStream', collection.type="doc", replication.type="global", map.type='json') (sensorId string, temperature double);
+```
+
+This example creates a sink to log events that arrive from a stream called `TemperatureOnlyStream` with the `temperature` attribute of type `double`:
+
+```sql
+CREATE SINK TemperatureOnlyStream WITH (type='stream', stream="TemperatureOnlyStream", replication.type="local", map.type='json') (temperature double);
+
+@info(name = 'Simple-selection')
+insert into TemperatureOnlyStream
+select temperature
+from TemperatureStream;
+```
+
+## Input
+
+When a JSON message is written to the collection `TemperatureStream`, it automatically gets mapped to an event in the `TemperatureStream` stream.
+
+```json
+{
+    "sensorId":"aq-14",
+    "temperature":35.4
+}
+```
+
+To process custom input messages, refer to Sinkmapper in [Functions](../query-guide/functions/index.md).
+
+## Output
+
+After processing, the event arriving at `TemperatureOnlyStream` will be emitted via `c8stream` sink.
```

Comment:
@diana-macrometa - I'm wandering whether we should not replace `c8stream` by `stream` or `gdn stream`? As far as I know `c8streams` is only for internal use.

Reply:
Except that it `c8local` and `c8local` are automatically added as a prefix for every stream that is created. I would prefer `gdn stream` or `stream`, because I have been told we should move away from `c8` wherever possible for branding reasons.

Which option is more correct, in your opinion? gdn stream or stream?

Reply:
I decided to go with stream, because we specify Macrometa in source, so that can be construed to include the sink.